src/Uno.UI/UI/Xaml/Internal/EventManager.uno.cs

```diff
@@ -0,0 +1,140 @@
+﻿#nullable enable
+
+#if UNO_HAS_ENHANCED_LIFECYCLE
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using Microsoft.UI.Xaml;
+using Uno.UI.Helpers;
+using Windows.Foundation;
+
+namespace Uno.UI.Xaml.Core;
+
+// Code here is mostly matching the logic from WinUI (even if the code isn't really a direct port), but it's in CoreImports.cpp and LayoutManager.cpp
+// Since we don't yet have these, and this code is very related to events, we add it to this EventManager partial.
+internal sealed partial class EventManager
+{
+	private readonly ConditionalWeakTable<FrameworkElement, object?> _layoutUpdatedSubscribers = new();
+	private readonly Queue<SizeChangedQueueItem> _sizeChangedQueue = new(1024);
+
+	internal bool HasPendingSizeChangedEvents => _sizeChangedQueue is { Count: > 0 };
+
+	private readonly struct SizeChangedQueueItem : IEquatable<SizeChangedQueueItem>
+	{
+		public FrameworkElement Element { get; }
+		public Size OldSize { get; }
+
+		public SizeChangedQueueItem(FrameworkElement element, Size oldSize)
+		{
+			Element = element;
+			OldSize = oldSize;
+		}
+
+		public override bool Equals(object? obj)
+			=> obj is SizeChangedQueueItem item && Equals(item);
+
+		public bool Equals(SizeChangedQueueItem other)
+			=> ReferenceEquals(Element, other.Element) && OldSize == other.OldSize;
+
+		public override int GetHashCode()
+			=> HashCode.Combine(Element, OldSize);
+
+		public static bool operator ==(SizeChangedQueueItem left, SizeChangedQueueItem right)
+			=> left.Equals(right);
+
+		public static bool operator !=(SizeChangedQueueItem left, SizeChangedQueueItem right)
+			=> !(left == right);
```

Comment:
Why not using a `record` here ?

Reply:
Records will use `IEqualityComparer<T>.Default`. In case of `Size` we don't implement `IEquatable<T>` so I think the default comparer will be `ObjectEqualityComparer` which will probably end up going through .NET's default value type equality which uses reflection (and some boxing will probably happen as well)

Reply:
As the `Size` is a `struct` I think it will instead compare its fields directly without any boxing nor reflection ... but I'm not sure neither so it makes sense to implement it by your own. Would be nice to add some comments to explain why you decided to implement the `IEquatable` here then ^^

Reply:
Yeah no reflection will happen as `Size` override `Equals`. But I think boxing will still happen because the `ObjectEqualityComparer` will end up calling `object.Equals(object)` method. I'll comment on this to clarify

Reply:
Added a comment as to why not using a record.